tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -312,12 +327,75 @@ private static void StartDiagnosticManager()
         }
 #endif
 
-        internal static void InitializeLiveDebugger()
+        private static void InitAsyncMethods(Tracer tracer)
```

Comment:
1. Perhaps we could improve the name of this method since it currently doesn't really describe what we're actually doing here (`AsyncMethods` is a C# implementation detail, not the purpose of the method we're trying to accomplish). `InitializeServicesInBackground`?
2. The name of `InitializeDiscoveryService` is also a bit problematic since we're not just initializing the service but also performing the actual discovery, perhaps we should move the latter part to this method?
3. It looks this code is structured this way because we have a dependency where we must first perform discovery and only then initialize `RcmPoller` and `LiveDebugger` iff we have the relevant endpoints for them, but that's wasn't immediately obvious from the code itself. We could solve that with a comment, but maybe better to try and making it more self-documenting by changing the signatures of `InitializeRcmPoller` and `InitializeLiveDebugger` so that it would receive the endpoint themselves, to make it more obvious?
4. The RFC states the following:

> The /info endpoint CAN also be used to check if RC is enabled. Regardless, tracers MUST continuously check for new configurations as RC could be enabled in the agent at any point in time.

I know we agreed we can leave this out in v0 and address this requirement in a future PR, but I'm wondering maybe the code would have been more readable/maintainable (and not much more difficult to implement) if we had implemented this now?